encoding/thrift/benchmark_test.go

```diff
@@ -0,0 +1,51 @@
+package thrift
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"go.uber.org/thriftrw/wire"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/api/transport/transporttest"
+)
+
+func BenchmarkThrift(b *testing.B) {
+	requestBody := wire.NewValueStruct(wire.Struct{})
+
+	mockCtrl := gomock.NewController(b)
+	defer mockCtrl.Finish()
+
+	proto := NewMockProtocol(mockCtrl)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	proto.EXPECT().DecodeEnveloped(gomock.Any()).Return(wire.Envelope{
+		Name:  "someMethod",
+		SeqID: 42,
+		Type:  wire.Exception,
+		Value: requestBody,
```

Comment:
possibly this should be better thought out as an end-to-end test, but I was trying to focus on a few key points.